src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,862 @@
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, name: newMethodIdentifier, parameters: validParameters, statements: newStatements.ToImmutable(), returnType: typeSymbol);
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var newMethodDeclaration = codeGenerationService.CreateMethodDeclaration(newMethod, options: new CodeGenerationOptions(options: options, parseOptions: expression.SyntaxTree.Options));
+            return newMethodDeclaration;
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a call to the method that introduced the parameter.
+        /// Example:
+        /// 
+        /// ***This is an intermediary step in which the original function has not be updated yet
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = [|x * y|];
+        /// }
+        /// 
+        /// ---------------------------------------------------->
+        /// 
+        /// public void M(int x, int y) // Generated overload
+        /// {
+        ///     M(x, y, x * y);
+        /// }
+        /// 
+        /// public void M(int x, int y) // Original function
+        /// {
+        ///     int f = x * y;
+        /// }
+        /// </summary>
+        private static async Task<SyntaxNode> GenerateNewMethodOverloadAsync(Document document, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, int insertionIndex, SyntaxGenerator generator, CancellationToken cancellationToken)
+        {
+            var named = methodSymbol.Parameters.Where(parameter => parameter.HasExplicitDefaultValue).Any();
+            var arguments = generator.CreateArguments(methodSymbol.Parameters);
+
+            // Remove trivia so the expression is in a single line and does not affect the spacing of the following line
+            arguments = arguments.Insert(insertionIndex, generator.Argument(expression.WithoutTrivia()));
+            var memberName = methodSymbol.IsGenericMethod
+                ? generator.GenericName(methodSymbol.Name, methodSymbol.TypeArguments)
+                : generator.IdentifierName(methodSymbol.Name);
+            var newStatements = ImmutableArray.CreateBuilder<SyntaxNode>();
+            var invocation = generator.InvocationExpression(memberName, arguments);
+
+            if (!methodSymbol.ReturnsVoid)
+            {
+                newStatements.Add(generator.ReturnStatement(invocation));
+            }
+            else
+            {
+                newStatements.Add(generator.ExpressionStatement(invocation));
+            }
+
+            var codeGenerationService = document.GetRequiredLanguageService<ICodeGenerationService>();
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, statements: newStatements.ToImmutable());
```

Comment:
```suggestion
            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, statements: SpecializedCollections.SingletonEnumerable(newStatement));
```

Reply:
requires an immutable array, is it better to use the CreateBuilder or create an IEnumerable and call ToImmutableArray on it?